workspaces/marketplace/plugins/marketplace/src/components/MarketplacePluginFilter.tsx

```diff
@@ -111,62 +118,131 @@ const SupportTypeFilter = () => {
   const [searchParams, setSearchParams] = useSearchParams();
   const pluginFacets = usePluginFacets({ facets: facetsKeys });
 
+  // Get all plugins data for exact count calculation
+  const marketplaceApi = useMarketplaceApi();
+  const allPluginsQuery = useQuery({
+    queryKey: ['marketplaceApi', 'getAllPlugins'],
+    queryFn: () => marketplaceApi.getPlugins({}),
+  });
+
   const facets = pluginFacets.data;
+  const allPlugins = useMemo(
+    () => allPluginsQuery.data?.items || [],
+    [allPluginsQuery.data],
+  );
+
+  // Single-pass exact count calculation
+  const exactCounts = useMemo(() => {
+    if (allPlugins.length === 0) {
+      return { ga: 0, community: 0, techPreview: 0, devPreview: 0 };
+    }
+
+    let ga = 0;
+    let community = 0;
+    let techPreview = 0;
+    let devPreview = 0;
+
+    allPlugins.forEach(plugin => {
+      const level = plugin.spec?.support?.level;
+      const name = plugin.spec?.support?.name;
+
+      if (level === SupportLevel.PRODUCTION && name === SupportProvider.RED_HAT)
+        ga++;
+      if (name === SupportProvider.BACKSTAGE_COMMUNITY) community++;
+      if (level === SupportLevel.TECH_PREVIEW) techPreview++;
+      if (level === SupportLevel.DEV_PREVIEW) devPreview++;
+    });
```

Comment:
if we fallback to the SupportLevel.CUSTOM if there is no level defined, and use "community" for all our plugins, we could simpliy use the facets results to count the categories instead of counting the entities manually.